src/lib.rs

```diff
@@ -690,19 +706,33 @@ pub trait HttpClient: Send + Sync + 'static {
     /// Send the given request and return the response
     fn request(
         &self,
-        req: Request<Body>,
-    ) -> Pin<Box<dyn Future<Output = hyper::Result<Response<Body>>> + Send>>;
+        req: Request<Full<Bytes>>,
+    ) -> Pin<
+        Box<
+            dyn Future<Output = Result<Response<Incoming>, hyper_util::client::legacy::Error>>
```

Comment:
I don't think it makes sense to have this very specific error appear in this trait. Maybe just make it `Error`?

Reply:
Sure let's do that then, and if we add the Other variant that gives more freedom for the implementer?